Migrate to dotnet standard

benbelow opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/softwire/lochnessbuilder/3)
<!-- Reviewable:end -->

dhedey commented:
Apparently I'm subscribed to this repo. Looks good.

Some quick thoughts:
* Are all the changes across all the cs files line ending changes? Are they intentional? If so, could these be put into a separate commit?
* What's the migration path for .NET Framework projects? Do they need to include `<Reference Include="netstandard" />` (next to eg `<Reference Include="System" />`) in their csproj file? If so, could we possibly mention that in the readme?
* Should some of the assembly info go into the new csproj? (I don't know about this, possibly not?)

benbelow commented:
> Are all the changes across all the cs files line ending changes?

Yes. 

> Are they intentional?

Not in the traditional sense. I have my global git settings set to `autocrlf=input` to ensure everything checked in is LF for non-windows users. 

As this is now .net standard I'd expect non-windows users to be able to maintain it, so I'd be happy to actually make the change to LF - but I don't mind if you disagree!

> What's the migration path for .NET Framework projects? Do they need to include <Reference Include="netstandard" /> (next to eg <Reference Include="System" />) in their csproj file? If so, could we possibly mention that in the readme?

I think that's right - I think in practice my package manager has handled this for me, but I agree it would be useful in the README!

> Should some of the assembly info go into the new csproj? (I don't know about this, possibly not?)

I'll have a look and move what seems sensible (other than version, which is already moved)

benbelow committed:
Migrate to dotnet standard

benbelow committed:
Change line endings to "LF"

Previously CRLF - but to enable maintainers on UNIX systems to work on the codebase without constantly changing the line endings, sticking to LF should make cross-platform maintenance smoother.

benbelow committed:
Add documentation detailing consumption from a .net framework project

benbelow committed:
Migrate additional properties from AssemblyInfo to csproj

dhedey commented on LochNessBuilder/LochNessBuilder.csproj:
```diff
@@ -1,8 +1,11 @@
 
     <PropertyGroup>
-        <TargetFramework>netstandard2.0</TargetFramework>
+        <TargetFramework>netstandard2.0</TargetFramework>]
```
Accidental bracket?

dhedey commented:
LF sounds good. All looks good, minus bonus bracket. Happy to approve.

But just realised I don't have write access. Thought I did (edit - yes - I somehow merged this PR: https://github.com/Softwire/LochNessBuilder/pull/2 - it must have changed since). Probably one for Helpdesk to give you / Jon A / someone else admin access (it was originally added to GitHub by an intern).

Then I guess we need to somehow publish it to NuGeT 😶

benbelow committed:
Remove accidental bracket

benbelow reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/softwire/lochnessbuilder/3)* status: 0 of 10 files reviewed, 1 unresolved discussion (waiting on @dhedey)

___
*[LochNessBuilder/LochNessBuilder.csproj, line 4 at r1](https://reviewable.io/reviews/softwire/lochnessbuilder/3#-LsbhW526mOE3hMO376z-r1-4:-LsbnMnM9znquw-34Mv6:b-896fix) ([raw file](https://github.com/softwire/lochnessbuilder/blob/f5f33e1572f8847d7754d489f64a4e8ed68900c4/LochNessBuilder/LochNessBuilder.csproj#L4)):*
<details><summary><i>Previously, dhedey (David Edey) wrote…</i></summary><blockquote>

Accidental bracket?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

benbelow reviewed (commented):
> Dismissed @dhedey from a discussion.
> *[Reviewable](https://reviewable.io/reviews/softwire/lochnessbuilder/3)* status: 0 of 10 files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

benbelow committed:
Add older target frameworks for seamless integration into .net framework projects

dhedey commented:
LGTM. Merge away when you're done :)

benbelow commented:
@dhedey do you know how this was release to nuget before? I can look into setting something up, but don't wnat to duplicate anything that's already in place

dhedey commented:
I've managed to turn "LochNess" user into an organization on NuGet, if you set up a NuGet account, can you let me know its username and I can add you as a LochNess admin
